app/services/base_api_manager.rb

```diff
@@ -7,28 +7,28 @@ def initialize(organization = "default")
   end
 
   #Fetches all the resources for the specific entity
-  def get_entities(entity_name = "", opts={})
+  def get_entities(entity_name="", opts={}, last_synchronization_date=nil, after_last_sync=false)
     begin
       batched_call = opts[:__skip] && opts[:__limit]
-      #the page requested is calculated from the offset ([:__skip)]) and (opts[:__limit])
-      if batched_call
-        page = opts[:__skip] == 0 ? 1 : (opts[:__skip] / opts[:__limit] + 1)
-        query = "page=#{page}&per_page=#{opts[:__limit]}"
-      end
+
+      query = QueryParamsManager.batched_call(opts) if batched_call
```

Comment:
You'll have an issue if it's both a batched_call and a date_filtered call (which should not happens right now but who knows what will happen in the future)